Horizon/Horizon/Sources/Features/Notebook/CourseList/ViewModel/NotebookViewModel.swift

```diff
@@ -0,0 +1,65 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import SwiftUI
+import Core
+
+@Observable final class NotebookViewModel {
+
+    var listItems: [NotebookListItem] = []
+
+    let router: Router
+
+    private let getCoursesInteractor: GetCoursesInteractor
+
+    private var getCoursesCancellable: AnyCancellable?
+
+    private var searchTextCancellable: AnyCancellable?
+
+    init(router: Router, getCoursesInteractor: GetCoursesInteractor) {
+        self.router = router
+        self.getCoursesInteractor = getCoursesInteractor
+
+        getCoursesCancellable = getCoursesInteractor.get().sink { _ in } receiveValue: { [weak self] courses in
+            self?.listItems = courses.map {
+                NotebookListItem(id: $0.id, course: $0.course, institution: $0.institution)
+            }
+        }
```

Comment:
For use cases like this, when you don't need to manage a stream's life cycle (compared to e.g a file download) you could have a property that collects and managed the subscriptions, then store each subscription's cancellable to that property. 

```
    private var subscriptions = Set<AnyCancellable>()
    
    then later on on the stream: 
        .sink()
        .store(in: &subscriptions)

```